Add attributes to AlarmDecoder

hawk259 opened the pull request:
Adding new attributes to AlarmDecoder: ac_power, backlight_on, battery_low, che…ck_zone, chime, entry_delay_off, programming_mode, ready, zone_bypassed

## Checklist:

If user exposed functionality or configuration variables are added/changed:
  - [ ] Documentation added/updated in [home-assistant.github.io](https://github.com/home-assistant/home-assistant.github.io)

If the code communicates with devices, web services, or third-party tools:
  - [ ] Local tests with `tox` run successfully. **Your PR cannot be merged unless tests pass**
  - [ ] New dependencies have been added to the `REQUIREMENTS` variable ([example][ex-requir]).
  - [ ] New dependencies are only imported inside functions that use them ([example][ex-import]).
  - [ ] New dependencies have been added to `requirements_all.txt` by running `script/gen_requirements_all.py`.
  - [ ] New files were added to `.coveragerc`.

If the code does not interact with devices:
  - [ ] Local tests with `tox` run successfully. **Your PR cannot be merged unless tests pass**
  - [ ] Tests have been added to verify that the new code works.

[ex-requir]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L14
[ex-import]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L54

fabaff committed:
Bump release to 0.60.0

hawk259 committed:
Add AlarmDecoder attributes: ac_power, backlight_on, battery_low, check_zone, chime, entry_delay_off, programming_mode, ready, zone_bypassed

hawk259 committed:
clear blank lines of white space

hawk259 committed:
clear blank lines of white space

hawk259 committed:
remove unused states

MartinHjelmare commented:
The alarmdecoder component with platforms use the home assistant asyncio api, but the alarmdecoder library doesn't seem to support asyncio. It would be good if you can first open a PR to fix the implementation by switching to the home assistant sync api. Or let me know if I'm totally mistaken and talking in my sleep, :smile:.

pvizeli committed:
Homematic next (#11156)

* Cleanup logic & New gen of HomeMatic

* fix lint

* cleanup

* fix coverage

* cleanup

* name consistenc

* fix lint

* Rename ip

* cleanup wrong property

* fix bug

* handle callback better

* fix lint

* Running now

pvizeli committed:
Homematic next (#11156)

* Cleanup logic & New gen of HomeMatic

* fix lint

* cleanup

* fix coverage

* cleanup

* name consistenc

* fix lint

* Rename ip

* cleanup wrong property

* fix bug

* handle callback better

* fix lint

* Running now

pvizeli committed:
Resolve hostnames (#11160)

pvizeli committed:
Resolve hostnames (#11160)

balloob committed:
Update frontend

balloob committed:
Update frontend

emlove committed:
Perform logbook filtering on the worker thread (#11161)

emlove committed:
Perform logbook filtering on the worker thread (#11161)

balloob committed:
Don't connect to cloud if subscription expired (#11163)

* Final touch for cloud component

* Fix test

balloob committed:
Don't connect to cloud if subscription expired (#11163)

* Final touch for cloud component

* Fix test

pvizeli committed:
Add install mode to homematic (#11164)

pvizeli committed:
Add install mode to homematic (#11164)

cmsimike committed:
Remove logging (#11173)

An error was being log that seems more like debug info

cmsimike committed:
Remove logging (#11173)

An error was being log that seems more like debug info

mtreinish committed:
Fix X10 commands for mochad light turn on (#11146)

* Fix X10 commands for mochad light turn on

This commit attempts to address issues that a lot of people are having
with the x10 light component. Originally this was written to use the
xdim (extended dim) X10 command. However, not every X10 dimmer device
supports the xdim command. Additionally, it turns out the number of
dim/brighness levels the X10 device supports is device specific and
there is no way to detect this (given the mostly 1 way nature of X10)

To address these issues, this commit removes the usage of xdim and
instead relies on using the 'on' command and the 'dim' command. This
should work on all x10 light devices. In an attempt to address the
different dim/brightness levels supported by different devices this
commit also adds a new optional config value, 'brightness_levels', to
specify if it's either 32, 64, or 256. By default 32 levels are used
as this is the normal case and what is documented by mochad.

Fixes #8943

* make code more readable

* fix style

* fix lint

* fix tests

mtreinish committed:
Fix X10 commands for mochad light turn on (#11146)

* Fix X10 commands for mochad light turn on

This commit attempts to address issues that a lot of people are having
with the x10 light component. Originally this was written to use the
xdim (extended dim) X10 command. However, not every X10 dimmer device
supports the xdim command. Additionally, it turns out the number of
dim/brighness levels the X10 device supports is device specific and
there is no way to detect this (given the mostly 1 way nature of X10)

To address these issues, this commit removes the usage of xdim and
instead relies on using the 'on' command and the 'dim' command. This
should work on all x10 light devices. In an attempt to address the
different dim/brightness levels supported by different devices this
commit also adds a new optional config value, 'brightness_levels', to
specify if it's either 32, 64, or 256. By default 32 levels are used
as this is the normal case and what is documented by mochad.

Fixes #8943

* make code more readable

* fix style

* fix lint

* fix tests

PhracturedBlue committed:
convert alarmdecoder interface from async to sync (#11168)

* convert alarmdecoder interface from async to sync

* Convert he rest of alarmdecoder rom async to sync

* Update alarmdecoder.py

* Update alarmdecoder.py

* Update alarmdecoder.py

* Update alarmdecoder.py

* Update alarmdecoder.py

* Update alarmdecoder.py

* Update alarmdecoder.py

* Update alarmdecoder.py

* Update alarmdecoder.py

* Update alarmdecoder.py

* Update alarmdecoder.py

* Update alarmdecoder.py

PhracturedBlue committed:
convert alarmdecoder interface from async to sync (#11168)

* convert alarmdecoder interface from async to sync

* Convert he rest of alarmdecoder rom async to sync

* Update alarmdecoder.py

* Update alarmdecoder.py

* Update alarmdecoder.py

* Update alarmdecoder.py

* Update alarmdecoder.py

* Update alarmdecoder.py

* Update alarmdecoder.py

* Update alarmdecoder.py

* Update alarmdecoder.py

* Update alarmdecoder.py

* Update alarmdecoder.py

* Update alarmdecoder.py

PeWu committed:
Try multiple methods of getting data in asuswrt. (#11140)

* Try multiple methods of getting data in asuswrt.

Solves #11108 and potentially #8112.

* fix style

* fix lint

rbdixon committed:
Revbump to SoCo 0.13 and add support for Night Sound and Speech Enhancement. (#10765)

Sonos Playbar and Playbase devices support Night Sound and Speech Enhancement
effects when playing from sources such as a TV. Adds a new service "sonos_set_option"
whichs accepts boolean options to control these audio features.

rbdixon committed:
Revbump to SoCo 0.13 and add support for Night Sound and Speech Enhancement. (#10765)

Sonos Playbar and Playbase devices support Night Sound and Speech Enhancement
effects when playing from sources such as a TV. Adds a new service "sonos_set_option"
whichs accepts boolean options to control these audio features.

fabaff committed:
Merge pull request #11153 from home-assistant/release-0-60

0.60

PhracturedBlue commented:
The async issue was addressed.  Can you fix your merge conflicts in this PR so it can get included?

fabaff committed:
Merge branch 'master' into dev

fabaff committed:
Bump dev to 0.61.0.dev0

maxlaverse committed:
Fix webdav calendar schema (#11185)

titilambert committed:
Hydroquebec component use now asyncio (#10795)

* Hydroquebec component use now asyncio

* Add tests

* Improve coverage

* fix tests

* Remove useless try/except and associated tests

Commit:
Hive Component Release Two (#11053)

* Add boost functionality to climate devices

* Update boost target temperature rounding

* Update with Colour Bulb Support

* colour bulb fix

* Requirements Update and colorsys import

* Add RGB Attribute - ATTR_RGB_COLOR

* Hive release-2

* add boost support for hive climate platform

* Colour Bulb - Varible update

* Boost - Tox error

* Convert colour to color

* Correct over indentation

* update version to 0.2.9 pyhiveapi

* Updated pyhiveapi to version 2.10 and altertered turn_n on fuction to 1 call

* Update climate doc string

* Update to is_aux_heat_on

* update to is_aux_heat_on

thibmaek committed:
Add Discogs Sensor platform (#10957)

* Add Discogs Sensor platform

* Add discogs module to requirements_all

* Fix wrong style var name

* PR Feedback (scan interval, mod. docstring)

* Added sensor.discogs to coveragerc

* Use SERVER_SOFTWARE helper for UA-String

* Don't setup platform if token is invalid

* Fix trailing whitespace for Hound CI

* Move client setup to setup()

markferry committed:
Fix statistics sensor mean and median when only one sample is available. (#11180)

* Fix statistics sensor mean and median when only one sample is available.

With only one data point stddev and variance throw an exception.
This would clear the (valid) mean and median calculations.

Separate the try..catch blocks for one-or-more and two-or-more stats so
that this doesn't happen.

Test this with a new sampling_size_1 test.

* test_statistics trivial whitespace fix

hawk259 committed:
fix conflict from async -> sync update

hawk259 committed:
Merge branch 'dev' of github.com:hawk259/home-assistant into dev

hawk259 committed:
move from STATE_UNKNOWN to None

PhracturedBlue commented:
I think you will need to rebase your changes on latest 'dev' branch otherwise the diff includes much more than just your change.  Tke a look at the 'files changed' tab to see the issue.

goyney commented:
@hawk259 I've rolled up your changes into https://github.com/home-assistant/home-assistant/issues/11271 so you don't need to do the rebase and added the required functionality to toggle the chime. If you close this PR, they can merge both of our changes together.

hawk259 commented:
@goyney Thank you!  Sorry I disappeared, got sick and it lasted way longer than I expected.
